### DISCUSSION TITLE
Add script sample viewer

### DISCUSSION THREAD

**Discussion by Adam-it** [Adam-it/cli-for-microsoft-365-extension#20] [Ideas]
I just got a really good idea to add some kind of viewer (all extend the current one with new section) to present script samples from CLI repo. 
Maybe the user could quickly view the sample and (by clicking on some button) create a local file with script sample contents as a starting point of their work 🤔

**Comment by pkbullock** [comment]
The PnP script samples repo injests some of the script samples from the CLI. I wrote a .net tool to process the format of one to the other. Garry Trinder approached me about a similar topic recently about how we communicate the best place for the samples.

**Reply by Adam-it** [reply]
With the CLI script samples I think we will add sample.json and also expose them to sample gallery (adoption page) this is along the way I guess 🤔🙂.
But this comment got me thinking that maybe this extension could also allow to view script samples from script sample repo and user, with one button click, could start off a new script file starting from a sample 🤔.
I like this idea.. I think I will promote this one to an issue and start working on it after I create a design for it.

**Comment by pkbullock** [comment]
If you do go for the sample.json route, I will need to find a way to disable the samples from PnP script samples also appearing in the unified sample gallery.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by Adam-it** [reply]
This is a bit off topic but I think we may recheck this topic on a short call with you and Garry (if I am not mistaken he is on vacation now so we'll need to wait 2 weeks 🙂).
If I remember well the main argument not to first sync samples from CLI repo to script sample repo and then to the solution gallery was the confusion we will make for the users when they from solution gallery first go to script samples and then to CLI repo. I think we agreed it would be better to direct the user to CLI repo directly from solution gallery when possible 🤔.
But we will still 'invest' effort in script samples repo as it creates great opportunity to show off the CLI and search for missing functionality 👍👍.
